client/v3/snapshot/v3_snapshot.go

```diff
@@ -94,9 +85,31 @@ func SaveWithVersion(ctx context.Context, lg *zap.Logger, cfg clientv3.Config, d
 		zap.String("etcd-version", resp.Version),
 	)
 
-	if err = os.Rename(partpath, dbPath); err != nil {
-		return resp.Version, fmt.Errorf("could not rename %s to %s (%v)", partpath, dbPath, err)
+	partPath := f.Name()
+	dbPath := strings.TrimSuffix(partPath, ".part")
+	if f != os.Stdout {
+		if err := os.Rename(partPath, dbPath); err != nil {
+			return resp.Version, fmt.Errorf("could not rename %s to %s (%v)", partPath, dbPath, err)
+		}
```

Comment:
We should move these to `SaveWithVersion`. 

`WriteSnapshotWithVersion` is a generic function, it shouldn't care about such details which only applies to normal `snapshot save` functionality.